addons/nuxeo-template-rendering/nuxeo-template-rendering-core/src/main/java/org/nuxeo/template/adapters/doc/TemplateBasedDocumentAdapterImpl.java

```diff
@@ -68,6 +69,15 @@
 
     protected final TemplateBindings bindings;
 
+    protected Serializer serializer;
+
+    public Serializer getSerializer() {
```

Comment:
Call it `getXmlSerializer`, and remove the `protected Serializer serializer` field caching, this is a premature optimization. Same in other classes with this pattern.